src/main/java/seedu/address/model/commitment/Lesson.java

```diff
@@ -63,20 +69,17 @@ public boolean equals(Object o) {
         if (this == o) {
             return true;
         }
-        if (o == null || getClass() != o.getClass()) {
+        if (!(o instanceof Lesson)) {
             return false;
         }
```

Comment:
Why remove the null check?

Reply:
Instanceof also checks for null. But more importantly, I changed this because the original `getClass()` way will force other to be *exactly* a `Lesson`, and it doesn't allow for inheritance.

Reply:
Dynamic binding also causes these to not work.